High level:
This notebook shows all the inconsistencies of field that were produced with dictionaries (and have hebrew in the name) with their respective numeric values for the markers_hebrew table.

The specific analysis below is based on data from 2020-01-13_views_and_main_tables folder from Jan 12, 2020 that can be found here: https://drive.google.com/drive/folders/1zrwfdkedy7jtRRsGi2dEn8zboZAG8_U2?usp=sharing

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
sns.set()
pd.options.display.max_rows = 300
pd.options.display.max_columns = 100

In [2]:
markers_raw = pd.read_csv('../../views_and_main_tables_2020_01/markers_hebrew.csv')
m_all = markers_raw[markers_raw['accident_year'] == 2019]

C:\Users\rkrichev\AppData\Local\Continuum\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3058: DtypeWarning: Columns (23,93) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [7]:
m_all.info(null_counts=True, max_cols=200)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 53030 entries, 4 to 799333
Data columns (total 101 columns):
id                                           53030 non-null int64
provider_and_id                              53030 non-null int64
provider_code                                53030 non-null int64
provider_code_hebrew                         53030 non-null object
file_type_police                             53030 non-null float64
accident_type                                53030 non-null int64
accident_type_hebrew                         53030 non-null object
accident_severity                            53030 non-null int64
accident_severity_hebrew                     53030 non-null object
accident_timestamp                           53030 non-null object
location_accuracy                            53030 non-null int64
location_accuracy_hebrew                     53030 non-null object
road_type                                    53030 non-null int64
road_type_hebrew        

### Helper functions

In [8]:
def calc_diff_counts_hebrew(data, feat_name):
    data = data[(data[feat_name].isnull() == False) & (data[feat_name + '_hebrew'].isnull() == False)]
    print(f'Shape of data: {data.shape}')
    return data[feat_name].value_counts().reset_index(drop=True) - \
           data[feat_name + '_hebrew'].value_counts().reset_index(drop=True)

def merge_with_hebrew(data, feat_name):
    nums_df = data[feat_name].value_counts().reset_index()
    nums_df.columns = ['index_' + feat_name, 'count']

    hebrew_df = data[feat_name + '_hebrew'].value_counts().reset_index()
    hebrew_df.columns = ['index_' + feat_name + '_hebrew', 'count']

    return pd.merge(nums_df, hebrew_df, how='outer', on='count')

### provider_code and provider_code_hebrew

In [10]:
# provider_code                                53030 non-null int64
# provider_code_hebrew                         53030 non-null object

In [11]:
calc_diff_counts_hebrew(m_all, 'provider_code')

Shape of data: (53030, 101)


0    0
1    0
dtype: int64

### accident_type and accident_type_hebrew

In [12]:
# accident_type                                53030 non-null int64
# accident_type_hebrew                         53030 non-null object

In [13]:
calc_diff_counts_hebrew(m_all, 'accident_type')

Shape of data: (53030, 101)


0     0
1     0
2     0
3     0
4     0
5     0
6     0
7     0
8     0
9     0
10    0
11    0
12    0
13    0
14    0
15    0
16    0
17    0
18    0
dtype: int64

### accident_severity and accident_severity_hebrew

In [15]:
# accident_severity                            53030 non-null int64
# accident_severity_hebrew                     53030 non-null object

In [16]:
calc_diff_counts_hebrew(m_all, 'accident_severity')

Shape of data: (53030, 101)


0    0
1    0
2    0
dtype: int64

In [17]:
### location_accuracy and location_accuracy_hebrew

In [18]:
# location_accuracy                            53030 non-null int64
# location_accuracy_hebrew                     53030 non-null object

In [19]:
calc_diff_counts_hebrew(m_all, 'location_accuracy')

Shape of data: (53030, 101)


0    0
1    0
2    0
3    0
4    0
dtype: int64

### road_type and road_type_hebrew

In [21]:
# road_type                                    53030 non-null int64
# road_type_hebrew                             53030 non-null object

In [22]:
calc_diff_counts_hebrew(m_all, 'road_type')

Shape of data: (53030, 101)


0    0
1    0
2    0
3    0
dtype: int64

### road_shape and road_shape_hebrew

In [23]:
# road_shape                                   53030 non-null int64
# road_shape_hebrew                            53030 non-null object

In [24]:
calc_diff_counts_hebrew(m_all, 'road_shape')

Shape of data: (53030, 101)


0    0
1    0
2    0
3    0
4    0
5    0
6    0
7    0
8    0
9    0
dtype: int64

### day_type and day_type_hebrew

In [26]:
# day_type                                     53030 non-null int64
# day_type_hebrew                              53030 non-null object

In [27]:
calc_diff_counts_hebrew(m_all, 'day_type')

Shape of data: (53030, 101)


0    0
1    0
2    0
dtype: int64

### police_unit and police_unit_hebrew

In [28]:
# police_unit                                  53030 non-null int64
# police_unit_hebrew                           53030 non-null object

In [29]:
calc_diff_counts_hebrew(m_all, 'police_unit')

Shape of data: (53030, 101)


0     0
1     0
2     0
3     0
4     0
5     0
6     0
7     0
8     0
9     0
10    0
11    0
12    0
dtype: int64

### one_lane and one_lane_hebrew

In [ ]:
# one_lane                                     53030 non-null int64
# one_lane_hebrew                              48614 non-null object

In [30]:
m_all[(m_all['one_lane'].isnull() == False) & (m_all['one_lane_hebrew'].isnull() == True)].shape

(4416, 101)

In [31]:
m_all[(m_all['one_lane'].isnull() == False) & (m_all['one_lane_hebrew'].isnull() == True)].describe()

id  provider_and_id  provider_code  file_type_police  \
count  4.416000e+03     4.416000e+03    4416.000000       4416.000000   
mean   2.019049e+09     1.224550e+10       1.022645          1.023098   
std    2.873457e+04     2.116299e+09       0.211630          0.213711   
min    2.019000e+09     1.201900e+10       1.000000          1.000000   
25%    2.019024e+09     1.201902e+10       1.000000          1.000000   
50%    2.019050e+09     1.201905e+10       1.000000          1.000000   
75%    2.019074e+09     1.201907e+10       1.000000          1.000000   
max    2.019100e+09     3.201910e+10       3.000000          3.000000   

       accident_type  accident_severity  location_accuracy    road_type  \
count    4416.000000        4416.000000        4416.000000  4416.000000   
mean        3.147418           2.819520           1.315897     2.204257   
std         2.837742           0.445732           0.725576     1.150288   
min         1.000000           1.000000           1.000000     1.000000   
25%         2.000000           3.000000           1.000000     1.000000   
50%         2.000000           3.000000           1.000000     2.000000   
75%         3.000000           3.000000           1.000000     3.000000   
max        20.000000           3.000000           9.000000     4.000000   

        road_shape     day_type  police_unit  one_lane   multi_lane  \
count  4416.000000  4416.000000  4416.000000    4416.0  4416.000000   
mean      8.310915     3.867980    34.170743       0.0     2.628170   
std       0.860103     0.560301    18.122893       0.0     0.797427   
min       1.000000     1.000000    11.000000       0.0     1.000000   
25%       8.000000     4.000000    14.000000       0.0     2.000000   
50%       8.000000     4.000000    38.000000       0.0     3.000000   
75%       9.000000     4.000000    51.000000       0.0     3.000000   
max      11.000000     4.000000    61.000000       0.0     5.000000   

       speed_limit  road_intactness   road_width    road_sign   road_light  \
count  4416.000000      4416.000000  4416.000000  4416.000000  4416.000000   
mean      2.436594         0.937274     3.000906     3.042799     2.075181   
std       1.811268         0.296306     1.169255     0.421643     1.886054   
min       0.000000         0.000000     0.000000     1.000000     1.000000   
25%       1.000000         1.000000     2.000000     3.000000     1.000000   
50%       1.000000         1.000000     3.000000     3.000000     1.000000   
75%       4.000000         1.000000     4.000000     3.000000     3.000000   
max       8.000000         4.000000     5.000000     5.000000    11.000000   

       road_control      weather  road_surface  road_object  object_distance  \
count   3159.000000  4416.000000   4416.000000  4416.000000      4416.000000   
mean       3.274454     1.199049      1.130661     8.692935         8.574728   
std        2.137681     1.060998      0.793748     1.237359         1.691451   
min        1.000000     1.000000      1.000000     1.000000         1.000000   
25%        2.000000     1.000000      1.000000     9.000000         9.000000   
50%        2.000000     1.000000      1.000000     9.000000         9.000000   
75%        6.000000     1.000000      1.000000     9.000000         9.000000   
max        7.000000     9.000000      9.000000     9.000000         9.000000   

       didnt_cross   cross_mode  cross_location  cross_direction        road1  \
count   4416.00000  4416.000000     4416.000000      4416.000000  1714.000000   
mean       8.95154     0.557292        0.537817         7.930254   349.498833   
std        0.43713     1.327939        1.255537         2.648725  1120.067266   
min        1.00000     0.000000        0.000000         1.000000     1.000000   
25%        9.00000     0.000000        0.000000         9.000000     5.000000   
50%        9.00000     0.000000        0.000000         9.000000    44.000000   
75%        9.00000     0.000000        0.000000     

In [33]:
m_all[m_all['one_lane'] == 0].shape

(4416, 101)

**Null Conclusion:** `one_lane` == 0 is missing

In [35]:
calc_diff_counts_hebrew(m_all, 'one_lane')

Shape of data: (48614, 101)


0    0
1    0
2    0
3    0
dtype: int64

### multi_lane and multi_lane_hebrew

In [ ]:
# multi_lane                                   53030 non-null int64
# multi_lane_hebrew                            4416 non-null object

In [36]:
m_all[(m_all['multi_lane'].isnull() == False) & (m_all['multi_lane_hebrew'].isnull() == True)].shape

(48614, 101)

In [37]:
m_all[(m_all['multi_lane'].isnull() == False) & (m_all['multi_lane_hebrew'].isnull() == True)].describe()

id  provider_and_id  provider_code  file_type_police  \
count  4.861400e+04     4.861400e+04   48614.000000      48614.000000   
mean   2.019050e+09     2.890513e+10       2.688608          2.702267   
std    3.063804e+04     7.251411e+09       0.725141          0.711921   
min    2.018012e+09     1.201801e+10       1.000000          1.000000   
25%    2.019025e+09     3.201901e+10       3.000000          3.000000   
50%    2.019050e+09     3.201904e+10       3.000000          3.000000   
75%    2.019075e+09     3.201907e+10       3.000000          3.000000   
max    2.019100e+09     3.201910e+10       3.000000          3.000000   

       accident_type  accident_severity  location_accuracy     road_type  \
count   48614.000000       48614.000000       48614.000000  48614.000000   
mean        4.791891           2.964702           1.907537      2.342946   
std         4.033328           0.203816           0.928229      0.885017   
min         1.000000           1.000000           1.000000      1.000000   
25%         3.000000           3.000000           1.000000      2.000000   
50%         3.000000           3.000000           2.000000      2.000000   
75%         5.000000           3.000000           3.000000      3.000000   
max        20.000000           3.000000           9.000000      4.000000   

         road_shape      day_type   police_unit      one_lane  multi_lane  \
count  48614.000000  48614.000000  48614.000000  48614.000000     48614.0   
mean       8.909861      3.894084     30.683486      8.141029         0.0   
std        0.418422      0.492964     17.858416      2.269380         0.0   
min        1.000000      1.000000     11.000000      1.000000         0.0   
25%        9.000000      4.000000     12.000000      9.000000         0.0   
50%        9.000000      4.000000     20.000000      9.000000         0.0   
75%        9.000000      4.000000     51.000000      9.000000         0.0   
max       11.000000      4.000000     61.000000      9.000000         0.0   

        speed_limit  road_intactness    road_width     road_sign  \
count  48614.000000     48614.000000  48614.000000  48614.000000   
mean       0.232711         0.121446      0.320052      4.745073   
std        0.785600         0.346452      0.905887      0.692575   
min        0.000000         0.000000      0.000000      1.000000   
25%        0.000000         0.000000      0.000000      5.000000   
50%        0.000000         0.000000      0.000000      5.000000   
75%        0.000000         0.000000      0.000000      5.000000   
max        8.000000         4.000000      5.000000      5.000000   

         road_light  road_control       weather  road_surface   road_object  \
count  48614.000000   3947.000000  48614.000000  48614.000000  48614.000000   
mean       8.840972      4.079301      8.001214      7.997367      8.908298   
std        3.346369      2.365177      2.635733      2.636790      0.671362   
min        1.000000      1.000000      1.000000      1.000000      1.000000   
25%        6.000000      2.000000      9.000000      9.000000      9.000000   
50%       11.000000      5.000000      9.000000      9.000000      9.000000   
75%       11.000000      7.000000      9.000000      9.000000      9.000000   
max       11.000000      7.000000      9.000000      9.000000      9.000000   

       object_distance   didnt_cross    cross_mode  cross_location  \
count     48614.000000  48614.000000  48614.000000    48614.000000   
mean          8.929588      8.979121      0.180051        0.110195   
std           0.709481      0.300296      0.919653        0.627615   
min           1.000000      1.000000      0.000000        0.000000   
25%           9.000000      9.000000      0.000000        0.000000   
50%           9.000000      9.000000      0.000000        0.000000   
75%           9.000000      9.000000      0.000000        0.000000   
max           9.000000      9.000000      9.000000        9.000000   

       cross_direction  

In [38]:
m_all[m_all['multi_lane'] == 0].shape

(48614, 101)

**Null Conclusion:** `multi_lane` == 0 is missing

In [39]:
calc_diff_counts_hebrew(m_all, 'multi_lane')

Shape of data: (4416, 101)


0    0
1    0
2    0
3    0
4    0
dtype: int64

### speed_limit and speed_limit_hebrew

In [40]:
# speed_limit                                  53030 non-null int64
# speed_limit_hebrew                           53030 non-null object

In [42]:
calc_diff_counts_hebrew(m_all, 'speed_limit')

Shape of data: (53030, 101)


0    0
1    0
2    0
3    0
4    0
5    0
6    0
7    0
8    0
dtype: int64

### road_intactness and road_intactness_hebrew

In [43]:
# road_intactness                              53030 non-null int64
# road_intactness_hebrew                       53030 non-null object

In [44]:
calc_diff_counts_hebrew(m_all, 'road_intactness')

Shape of data: (53030, 101)


0    0
1    0
2    0
3    0
4    0
dtype: int64

### road_width and road_width_hebrew

In [46]:
# road_width                                   53030 non-null int64
# road_width_hebrew                            53030 non-null object

In [47]:
calc_diff_counts_hebrew(m_all, 'road_width')

Shape of data: (53030, 101)


0    0
1    0
2    0
3    0
4    0
5    0
dtype: int64

### road_sign and road_sign_hebrew

In [ ]:
# road_sign                                    53030 non-null int64
# road_sign_hebrew                             53030 non-null object

In [49]:
calc_diff_counts_hebrew(m_all, 'road_sign')

Shape of data: (53030, 101)


0    0
1    0
2    0
3    0
4    0
dtype: int64

### road_light and road_light_hebrew

In [50]:
# road_light                                   53030 non-null int64
# road_light_hebrew                            53030 non-null object

In [51]:
calc_diff_counts_hebrew(m_all, 'road_light')

Shape of data: (53030, 101)


0    0
1    0
2    0
3    0
4    0
5    0
6    0
7    0
8    0
dtype: int64

### road_control and road_control_hebrew

In [ ]:
# road_control                                 7106 non-null float64
# road_control_hebrew                          7106 non-null object

In [53]:
calc_diff_counts_hebrew(m_all, 'road_control')

Shape of data: (7106, 101)


0    0
1    0
2    0
3    0
4    0
5    0
dtype: int64

### weather and weather_hebrew 

In [ ]:
# weather                                      53030 non-null int64
# weather_hebrew                               53030 non-null object

In [54]:
calc_diff_counts_hebrew(m_all, 'weather')

Shape of data: (53030, 101)


0    0
1    0
2    0
3    0
4    0
5    0
dtype: int64

### road_surface and road_surface_hebrew

In [55]:
# road_surface                                 53030 non-null int64
# road_surface_hebrew                          53030 non-null object

In [56]:
calc_diff_counts_hebrew(m_all, 'road_surface')

Shape of data: (53030, 101)


0    0
1    0
2    0
3    0
4    0
5    0
6    0
dtype: int64

### road_object and road_object_hebrew

In [57]:
# road_object                                  53030 non-null int64
# road_object_hebrew                           53030 non-null object

In [58]:
calc_diff_counts_hebrew(m_all, 'road_object')

Shape of data: (53030, 101)


0    0
1    0
2    0
3    0
4    0
5    0
6    0
7    0
8    0
dtype: int64

### object_distance and object_distance_hebrew

In [ ]:
# object_distance                              53030 non-null int64
# object_distance_hebrew                       53030 non-null object

In [60]:
calc_diff_counts_hebrew(m_all, 'object_distance')

Shape of data: (53030, 101)


0    0
1    0
2    0
3    0
4    0
5    0
dtype: int64

### didnt_cross and didnt_cross_hebrew

In [61]:
# didnt_cross                                  53030 non-null int64
# didnt_cross_hebrew                           53030 non-null object

In [62]:
calc_diff_counts_hebrew(m_all, 'didnt_cross')

Shape of data: (53030, 101)


0    0
1    0
2    0
3    0
4    0
5    0
6    0
7    0
dtype: int64

### cross_mode and cross_mode_hebrew

In [63]:
# cross_mode                                   53030 non-null int64
# cross_mode_hebrew                            3185 non-null object

In [64]:
m_all[(m_all['cross_mode'].isnull() == False) & (m_all['cross_mode_hebrew'].isnull() == True)].shape

(49845, 101)

In [65]:
m_all[(m_all['cross_mode'].isnull() == False) & (m_all['cross_mode_hebrew'].isnull() == True)].describe()

id  provider_and_id  provider_code  file_type_police  \
count  4.984500e+04     4.984500e+04   49845.000000      49845.000000   
mean   2.019050e+09     2.819600e+10       2.617695          2.630454   
std    3.055879e+04     7.864258e+09       0.786426          0.776234   
min    2.018012e+09     1.201801e+10       1.000000          1.000000   
25%    2.019025e+09     3.201901e+10       3.000000          3.000000   
50%    2.019050e+09     3.201904e+10       3.000000          3.000000   
75%    2.019075e+09     3.201907e+10       3.000000          3.000000   
max    2.019100e+09     3.201910e+10       3.000000          3.000000   

       accident_type  accident_severity  location_accuracy     road_type  \
count   49845.000000       49845.000000       49845.000000  49845.000000   
mean        4.883138           2.961641           1.874972      2.371813   
std         3.983878           0.214094           0.929663      0.910533   
min         1.000000           1.000000           1.000000      1.000000   
25%         3.000000           3.000000           1.000000      2.000000   
50%         3.000000           3.000000           2.000000      2.000000   
75%         5.000000           3.000000           3.000000      3.000000   
max        20.000000           3.000000           9.000000      4.000000   

         road_shape      day_type   police_unit      one_lane    multi_lane  \
count  49845.000000  49845.000000  49845.000000  49845.000000  49845.000000   
mean       8.878604      3.890541     30.871381      7.697362      0.189126   
std        0.482176      0.501958     17.909484      2.935151      0.709431   
min        1.000000      1.000000     11.000000      0.000000      0.000000   
25%        9.000000      4.000000     12.000000      9.000000      0.000000   
50%        9.000000      4.000000     20.000000      9.000000      0.000000   
75%        9.000000      4.000000     51.000000      9.000000      0.000000   
max       11.000000      4.000000     61.000000      9.000000      5.000000   

        speed_limit  road_intactness    road_width     road_sign  \
count  49845.000000     49845.000000  49845.000000  49845.000000   
mean       0.393861         0.160498      0.476156      4.665262   
std        1.115435         0.383797      1.144563      0.767800   
min        0.000000         0.000000      0.000000      1.000000   
25%        0.000000         0.000000      0.000000      5.000000   
50%        0.000000         0.000000      0.000000      5.000000   
75%        0.000000         0.000000      0.000000      5.000000   
max        8.000000         4.000000      5.000000      5.000000   

         road_light  road_control       weather  road_surface   road_object  \
count  49845.000000   5640.000000  49845.000000  49845.000000  49845.000000   
mean       8.522560      3.584574      7.676698      7.668813      8.885204   
std        3.584579      2.198343      2.961477      2.966843      0.753202   
min        1.000000      1.000000      1.000000      1.000000      1.000000   
25%        6.000000      2.000000      9.000000      9.000000      9.000000   
50%       11.000000      2.000000      9.000000      9.000000      9.000000   
75%       11.000000      6.000000      9.000000      9.000000      9.000000   
max       11.000000      7.000000      9.000000      9.000000      9.000000   

       object_distance   didnt_cross  cross_mode  cross_location  \
count     49845.000000  49845.000000     49845.0    49845.000000   
mean          8.896218      8.993680         0.0        0.000782   
std           0.857752      0.140862         0.0        0.057180   
min           1.000000      1.000000         0.0        0.000000   
25%           9.000000      9.000000         0.0        0.000000   
50%           9.000000      9.000000         0.0        0.000000   
75%           9.000000      9.000000         0.0        0.000000   
max           9.000000      9.000000         0.0        9.000000   

       cross_direction  

In [66]:
m_all[m_all['cross_mode']==0].shape

(49845, 101)

**Null Conclusions**: All `cross_mode` == 0 translation is missing

In [67]:
calc_diff_counts_hebrew(m_all, 'cross_mode')

Shape of data: (3185, 101)


0    0
1    0
2    0
3    0
4    0
5    0
dtype: int64

### cross_location and cross_location_hebrew

In [68]:
# cross_location                               53030 non-null int64
# cross_location_hebrew                        2545 non-null object

In [69]:
m_all[(m_all['cross_location'].isnull() == False) & (m_all['cross_location_hebrew'].isnull() == True)].shape

(50485, 101)

In [70]:
m_all[(m_all['cross_location'].isnull() == False) & (m_all['cross_location_hebrew'].isnull() == True)].describe()

id  provider_and_id  provider_code  file_type_police  \
count  5.048500e+04     5.048500e+04   50485.000000      50485.000000   
mean   2.019050e+09     2.817316e+10       2.615411          2.628206   
std    3.054865e+04     7.882146e+09       0.788215          0.778054   
min    2.018012e+09     1.201801e+10       1.000000          1.000000   
25%    2.019025e+09     3.201901e+10       3.000000          3.000000   
50%    2.019050e+09     3.201904e+10       3.000000          3.000000   
75%    2.019075e+09     3.201907e+10       3.000000          3.000000   
max    2.019100e+09     3.201910e+10       3.000000          3.000000   

       accident_type  accident_severity  location_accuracy     road_type  \
count   50485.000000       50485.000000       50485.000000  50485.000000   
mean        4.837140           2.960464           1.876102      2.367475   
std         3.983727           0.217722           0.928747      0.907393   
min         1.000000           1.000000           1.000000      1.000000   
25%         3.000000           3.000000           1.000000      2.000000   
50%         3.000000           3.000000           2.000000      2.000000   
75%         5.000000           3.000000           3.000000      3.000000   
max        20.000000           3.000000           9.000000      4.000000   

         road_shape      day_type   police_unit      one_lane    multi_lane  \
count  50485.000000  50485.000000  50485.000000  50485.000000  50485.000000   
mean       8.878063      3.890641     30.884877      7.690383      0.188888   
std        0.482056      0.501863     17.896841      2.939939      0.709523   
min        1.000000      1.000000     11.000000      0.000000      0.000000   
25%        9.000000      4.000000     12.000000      9.000000      0.000000   
50%        9.000000      4.000000     20.000000      9.000000      0.000000   
75%        9.000000      4.000000     51.000000      9.000000      0.000000   
max       11.000000      4.000000     61.000000      9.000000      5.000000   

        speed_limit  road_intactness    road_width     road_sign  \
count  50485.000000     50485.000000  50485.000000  50485.000000   
mean       0.393067         0.161652      0.477310      4.663365   
std        1.111878         0.385688      1.144338      0.769639   
min        0.000000         0.000000      0.000000      1.000000   
25%        0.000000         0.000000      0.000000      5.000000   
50%        0.000000         0.000000      0.000000      5.000000   
75%        0.000000         0.000000      0.000000      5.000000   
max        8.000000         4.000000      5.000000      5.000000   

         road_light  road_control       weather  road_surface   road_object  \
count  50485.000000   5686.000000  50485.000000  50485.000000  50485.000000   
mean       8.515084      3.588815      7.668377      7.660513      8.885055   
std        3.590957      2.204363      2.968998      2.974160      0.754880   
min        1.000000      1.000000      1.000000      1.000000      1.000000   
25%        6.000000      2.000000      9.000000      9.000000      9.000000   
50%       11.000000      2.000000      9.000000      9.000000      9.000000   
75%       11.000000      6.000000      9.000000      9.000000      9.000000   
max       11.000000      7.000000      9.000000      9.000000      9.000000   

       object_distance   didnt_cross    cross_mode  cross_location  \
count     50485.000000  50485.000000  50485.000000         50485.0   
mean          8.895276      8.975834      0.083708             0.0   
std           0.861718      0.320833      0.771725             0.0   
min           1.000000      1.000000      0.000000             0.0   
25%           9.000000      9.000000      0.000000             0.0   
50%           9.000000      9.000000      0.000000             0.0   
75%           9.000000      9.000000      0.000000             0.0   
max           9.000000      9.000000      9.000000             0.0   

      

In [71]:
m_all[m_all['cross_location']==0].shape

(50485, 101)

**Null Conclusion:** all `cross_location` == 0 has no translation

In [72]:
calc_diff_counts_hebrew(m_all, 'cross_location')

Shape of data: (2545, 101)


0    0
1    0
2    0
3    0
4    0
dtype: int64

### cross_direction and cross_direction_hebrew

In [73]:
# cross_direction                              53030 non-null int64
# cross_direction_hebrew                       53030 non-null object

In [74]:
calc_diff_counts_hebrew(m_all, 'cross_direction')

Shape of data: (53030, 101)


0    0
1    0
2    0
dtype: int64

In [75]:
### geo_area and geo_area_hebrew

In [77]:
# geo_area                                     53030 non-null int64
# geo_area_hebrew                              53030 non-null object

In [78]:
calc_diff_counts_hebrew(m_all, 'geo_area')

Shape of data: (53030, 101)


0    0
1    0
dtype: int64

### day_night and day_night_hebrew

In [79]:
# day_night                                    53030 non-null int64
# day_night_hebrew                             53030 non-null object

In [80]:
calc_diff_counts_hebrew(m_all, 'day_night')

Shape of data: (53030, 101)


0    0
1    0
dtype: int64

### day_in_week and day_in_week_hebrew

In [81]:
# day_in_week                                  53030 non-null int64
# day_in_week_hebrew                           53030 non-null object

In [82]:
calc_diff_counts_hebrew(m_all, 'day_in_week')

Shape of data: (53030, 101)


0    0
1    0
2    0
3    0
4    0
5    0
6    0
dtype: int64

### traffic_light and traffic_light_hebrew

In [83]:
# traffic_light                                53030 non-null int64
# traffic_light_hebrew                         53030 non-null object

In [84]:
calc_diff_counts_hebrew(m_all, 'traffic_light')

Shape of data: (53030, 101)


0    0
1    0
2    0
dtype: int64

### region and region_hebrew

In [85]:
# region                                       42898 non-null float64
# region_hebrew                                42898 non-null object

In [86]:
calc_diff_counts_hebrew(m_all, 'region')

Shape of data: (42898, 101)


0    0
1    0
2    0
3    0
4    0
5    0
6    0
7    0
dtype: int64

### district and district_hebrew

In [87]:
# district                                     53030 non-null int64
# district_hebrew                              38921 non-null object

In [88]:
m_all[(m_all['district'].isnull() == False) & (m_all['district_hebrew'].isnull() == True)].shape

(14109, 101)

In [89]:
m_all[(m_all['district'].isnull() == False) & (m_all['district_hebrew'].isnull() == True)].describe()

id  provider_and_id  provider_code  file_type_police  \
count  1.410900e+04     1.410900e+04   14109.000000      14109.000000   
mean   2.019050e+09     2.638151e+10       2.436246          2.449429   
std    3.342239e+04     8.998592e+09       0.899859          0.893347   
min    2.018012e+09     1.201801e+10       1.000000          1.000000   
25%    2.019025e+09     1.201909e+10       1.000000          1.000000   
50%    2.019051e+09     3.201903e+10       3.000000          3.000000   
75%    2.019075e+09     3.201907e+10       3.000000          3.000000   
max    2.019100e+09     3.201910e+10       3.000000          3.000000   

       accident_type  accident_severity  location_accuracy     road_type  \
count   14109.000000       14109.000000       14109.000000  14109.000000   
mean        4.368843           2.926501           1.130838      3.694096   
std         3.316353           0.304812           0.766949      0.461113   
min         1.000000           1.000000           1.000000      2.000000   
25%         3.000000           3.000000           1.000000      3.000000   
50%         3.000000           3.000000           1.000000      4.000000   
75%         4.000000           3.000000           1.000000      4.000000   
max        20.000000           3.000000           9.000000      4.000000   

         road_shape      day_type   police_unit      one_lane    multi_lane  \
count  14109.000000  14109.000000  14109.000000  14109.000000  14109.000000   
mean       8.803955      3.875824     30.572046      7.011269      0.305266   
std        0.676143      0.534495     17.486237      3.467979      0.854789   
min        1.000000      1.000000     11.000000      0.000000      0.000000   
25%        9.000000      4.000000     12.000000      3.000000      0.000000   
50%        9.000000      4.000000     20.000000      9.000000      0.000000   
75%        9.000000      4.000000     51.000000      9.000000      0.000000   
max       11.000000      4.000000     61.000000      9.000000      5.000000   

        speed_limit  road_intactness    road_width     road_sign  \
count  14109.000000     14109.000000  14109.000000  14109.000000   
mean       0.980084         0.241831      0.800411      4.503012   
std        1.836710         0.443495      1.466572      0.884975   
min        0.000000         0.000000      0.000000      1.000000   
25%        0.000000         0.000000      0.000000      5.000000   
50%        0.000000         0.000000      0.000000      5.000000   
75%        1.000000         0.000000      0.000000      5.000000   
max        8.000000         4.000000      5.000000      5.000000   

         road_light  road_control       weather  road_surface   road_object  \
count  14109.000000   2062.000000  14109.000000  14109.000000  14109.000000   
mean       7.786803      3.388458      7.025799      7.005812      8.839110   
std        3.923376      2.177456      3.432597      3.443121      0.816386   
min        1.000000      1.000000      1.000000      1.000000      1.000000   
25%        6.000000      2.000000      9.000000      2.000000      9.000000   
50%       11.000000      2.000000      9.000000      9.000000      9.000000   
75%       11.000000      6.000000      9.000000      9.000000      9.000000   
max       11.000000      7.000000      9.000000      9.000000      9.000000   

       object_distance  didnt_cross    cross_mode  cross_location  \
count     14109.000000  14109.00000  14109.000000     14109.00000   
mean          8.820469      8.98866      0.033808         0.01942   
std           1.116741      0.22719      0.418497         0.26606   
min           1.000000      1.00000      0.000000         0.00000   
25%           9.000000      9.00000      0.000000         0.00000   
50%           9.000000      9.00000      0.000000         0.00000   
75%           9.000000      9.00000      0.000000         0.00000   
max           9.000000      9.00000      9.000000         9.00000   

       cross_di

In [91]:
m_all[m_all['district'] == 99].shape

(14109, 101)

**Null Conclusion:** all `district` == 99 missing translation

In [92]:
calc_diff_counts_hebrew(m_all, 'district')

Shape of data: (38921, 101)


0     0
1     0
2     0
3     0
4     0
5     0
6     0
7     0
8     0
9     0
10    0
11    0
12    0
13    0
14    0
15    0
16    0
17    0
18    0
19    0
20    0
21    0
dtype: int64

### natural_area and natural_area_hebrew

In [93]:
# natural_area                                 42898 non-null float64
# natural_area_hebrew                          38822 non-null object

In [94]:
m_all[(m_all['natural_area'].isnull() == False) & (m_all['natural_area_hebrew'].isnull() == True)].shape

(4076, 101)

In [95]:
m_all[(m_all['natural_area'].isnull() == False) & (m_all['natural_area_hebrew'].isnull() == True)].describe()

id  provider_and_id  provider_code  file_type_police  \
count  4.076000e+03     4.076000e+03         4076.0       4076.000000   
mean   2.019049e+09     1.201905e+10            1.0          1.050540   
std    3.654600e+04     3.654600e+04            0.0          0.313926   
min    2.018012e+09     1.201801e+10            1.0          1.000000   
25%    2.019025e+09     1.201903e+10            1.0          1.000000   
50%    2.019050e+09     1.201905e+10            1.0          1.000000   
75%    2.019075e+09     1.201907e+10            1.0          1.000000   
max    2.019100e+09     1.201910e+10            1.0          3.000000   

       accident_type  accident_severity  location_accuracy    road_type  \
count    4076.000000        4076.000000        4076.000000  4076.000000   
mean        3.921246           2.738960           1.112120     3.577772   
std         3.017158           0.528977           0.587343     0.569220   
min         1.000000           1.000000           1.000000     1.000000   
25%         2.000000           3.000000           1.000000     3.000000   
50%         3.000000           3.000000           1.000000     4.000000   
75%         5.000000           3.000000           1.000000     4.000000   
max        20.000000           3.000000           9.000000     4.000000   

        road_shape     day_type  police_unit     one_lane   multi_lane  \
count  4076.000000  4076.000000  4076.000000  4076.000000  4076.000000   
mean      8.319921     3.846909    28.906035     2.113837     1.059863   
std       1.126678     0.599483    17.325856     2.806879     1.319732   
min       1.000000     1.000000    11.000000     0.000000     0.000000   
25%       8.000000     4.000000    12.000000     0.000000     0.000000   
50%       8.000000     4.000000    20.000000     1.000000     0.000000   
75%       9.000000     4.000000    51.000000     3.000000     2.000000   
max      11.000000     4.000000    61.000000     9.000000     5.000000   

       speed_limit  road_intactness   road_width    road_sign   road_light  \
count  4076.000000      4076.000000  4076.000000  4076.000000  4076.000000   
mean      3.341021         0.838567     2.771099     3.274779     2.965162   
std       1.882694         0.428373     1.404937     0.773526     3.028902   
min       0.000000         0.000000     0.000000     1.000000     1.000000   
25%       1.000000         1.000000     2.000000     3.000000     1.000000   
50%       4.000000         1.000000     3.000000     3.000000     1.000000   
75%       5.000000         1.000000     4.000000     3.000000     3.000000   
max       8.000000         4.000000     5.000000     5.000000    11.000000   

       road_control      weather  road_surface  road_object  object_distance  \
count   2025.000000  4076.000000   4076.000000  4076.000000      4076.000000   
mean       3.456790     2.149411      2.081943     8.676644         8.510304   
std        2.207433     2.736170      2.660257     1.171630         1.799834   
min        1.000000     1.000000      1.000000     1.000000         1.000000   
25%        2.000000     1.000000      1.000000     9.000000         9.000000   
50%        2.000000     1.000000      1.000000     9.000000         9.000000   
75%        6.000000     1.000000      1.000000     9.000000         9.000000   
max        7.000000     9.000000      9.000000     9.000000         9.000000   

       didnt_cross   cross_mode  cross_location  cross_direction        road1  \
count  4076.000000  4076.000000     4076.000000      4076.000000  3973.000000   
mean      8.963690     0.113101        0.082434         8.807164   751.723886   
std       0.402027     0.732231        0.523392         1.194510  1889.794208   
min       1.000000     0.000000        0.000000         1.000000     1.000000   
25%       9.000000     0.000000        0.000000         9.000000    20.000000   
50%       9.000000     0.000000        0.000000         9.000000    70.000000   
75%       9.000000     0.

In [96]:
m_all[(m_all['natural_area'].isnull() == False) & (m_all['natural_area_hebrew'].isnull() == True)]['natural_area'].value_counts()

999.0    3977
777.0      99
Name: natural_area, dtype: int64

**Null Conclusion:** Following `natural_area` had translation issues:
- 999 - all values missing
- 777 - all values missing

In [97]:
calc_diff_counts_hebrew(m_all, 'natural_area')

Shape of data: (38822, 101)


0     0
1     0
2     0
3     0
4     0
5     0
6     0
7     0
8     0
9     0
10    0
11    0
12    0
13    0
14    0
15    0
16    0
17    0
18    0
19    0
20    0
21    0
22    0
23    0
24    0
25    0
26    0
27    0
28    0
29    0
30    0
31    0
32    0
33    0
34    0
35    0
36    0
37    0
38    0
39    0
40    0
41    0
42    0
43    0
44    0
45    0
46    0
47    0
48    0
49    0
50    0
51    0
52    0
53    0
54    0
55    0
56    0
57    0
dtype: int64

### municipal_status and municipal_status_hebrew

In [98]:
# municipal_status                             42805 non-null float64
# municipal_status_hebrew                      42805 non-null object

In [99]:
calc_diff_counts_hebrew(m_all, 'municipal_status')

Shape of data: (42805, 101)


0     0
1     0
2     0
3     0
4     0
5     0
6     0
7     0
8     0
9     0
10    0
11    0
12    0
13    0
14    0
15    0
16    0
17    0
18    0
19    0
20    0
21    0
22    0
23    0
24    0
25    0
26    0
27    0
28    0
29    0
30    0
31    0
32    0
33    0
34    0
35    0
36    0
37    0
38    0
39    0
40    0
41    0
42    0
43    0
44    0
45    0
46    0
47    0
48    0
49    0
50    0
51    0
52    0
53    0
54    0
dtype: int64

### yishuv_shape and yishuv_shape_hebrew

In [ ]:
# yishuv_shape                                 42898 non-null float64
# yishuv_shape_hebrew                          38921 non-null object

In [100]:
m_all[(m_all['yishuv_shape'].isnull() == False) & (m_all['yishuv_shape_hebrew'].isnull() == True)].shape

(3977, 101)

In [101]:
m_all[(m_all['yishuv_shape'].isnull() == False) & (m_all['yishuv_shape_hebrew'].isnull() == True)].describe()

id  provider_and_id  provider_code  file_type_police  \
count  3.977000e+03     3.977000e+03         3977.0       3977.000000   
mean   2.019049e+09     1.201905e+10            1.0          1.046769   
std    3.671299e+04     3.671299e+04            0.0          0.302280   
min    2.018012e+09     1.201801e+10            1.0          1.000000   
25%    2.019025e+09     1.201903e+10            1.0          1.000000   
50%    2.019050e+09     1.201905e+10            1.0          1.000000   
75%    2.019075e+09     1.201907e+10            1.0          1.000000   
max    2.019100e+09     1.201910e+10            1.0          3.000000   

       accident_type  accident_severity  location_accuracy    road_type  \
count    3977.000000        3977.000000        3977.000000  3977.000000   
mean        3.929847           2.739251           1.097058     3.625094   
std         2.993346           0.529939           0.571484     0.484159   
min         1.000000           1.000000           1.000000     3.000000   
25%         2.000000           3.000000           1.000000     3.000000   
50%         3.000000           3.000000           1.000000     4.000000   
75%         5.000000           3.000000           1.000000     4.000000   
max        20.000000           3.000000           9.000000     4.000000   

        road_shape     day_type  police_unit     one_lane   multi_lane  \
count  3977.000000  3977.000000  3977.000000  3977.000000  3977.000000   
mean      8.316067     3.845361    28.573799     2.086749     1.069399   
std       1.128970     0.602622    17.409500     2.791779     1.322005   
min       1.000000     1.000000    11.000000     0.000000     0.000000   
25%       8.000000     4.000000    12.000000     0.000000     0.000000   
50%       8.000000     4.000000    20.000000     1.000000     0.000000   
75%       9.000000     4.000000    51.000000     3.000000     2.000000   
max      11.000000     4.000000    61.000000     9.000000     5.000000   

       speed_limit  road_intactness   road_width    road_sign   road_light  \
count  3977.000000      3977.000000  3977.000000  3977.000000  3977.000000   
mean      3.399799         0.840835     2.784762     3.273573     2.954488   
std       1.863577         0.423840     1.404913     0.768853     3.008655   
min       0.000000         0.000000     0.000000     1.000000     1.000000   
25%       2.000000         1.000000     2.000000     3.000000     1.000000   
50%       4.000000         1.000000     3.000000     3.000000     1.000000   
75%       5.000000         1.000000     4.000000     3.000000     3.000000   
max       8.000000         4.000000     5.000000     5.000000    11.000000   

       road_control      weather  road_surface  road_object  object_distance  \
count   1991.000000  3977.000000   3977.000000  3977.000000      3977.000000   
mean       3.420894     2.138798      2.070153     8.671612         8.502892   
std        2.190580     2.724497      2.646009     1.178821         1.812808   
min        1.000000     1.000000      1.000000     1.000000         1.000000   
25%        2.000000     1.000000      1.000000     9.000000         9.000000   
50%        2.000000     1.000000      1.000000     9.000000         9.000000   
75%        6.000000     1.000000      1.000000     9.000000         9.000000   
max        7.000000     9.000000      9.000000     9.000000         9.000000   

       didnt_cross   cross_mode  cross_location  cross_direction        road1  \
count  3977.000000  3977.000000     3977.000000      3977.000000  3973.000000   
mean      8.964295     0.092784        0.057581         8.867488   751.723886   
std       0.401497     0.698655        0.440588         0.992320  1889.794208   
min       1.000000     0.000000        0.000000         1.000000     1.000000   
25%       9.000000     0.000000        0.000000         9.000000    20.000000   
50%       9.000000     0.000000        0.000000         9.000000    70.000000   
75%       9.000000     0.

In [102]:
m_all[m_all['yishuv_shape']==99].shape

(3977, 101)

**Null Conclusion:** `yishuv_shape` translation issues: 99

In [104]:
calc_diff_counts_hebrew(m_all, 'yishuv_shape')

Shape of data: (38921, 101)


0     0
1     0
2     0
3     0
4     0
5     0
6     0
7     0
8     0
9     0
10    0
11    0
12    0
13    0
14    0
15    0
16    0
17    0
18    0
19    0
20    0
21    0
22    0
23    0
24    0
25    0
dtype: int64

### street1 and street1_hebrew

In [105]:
# street1                                      28750 non-null float64
# street1_hebrew                               27712 non-null object

In [106]:
m_all[(m_all['street1'].isnull() == False) & (m_all['street1_hebrew'].isnull() == True)].shape

(1038, 101)

In [107]:
m_all[(m_all['street1'].isnull() == False) & (m_all['street1_hebrew'].isnull() == True)].describe()

id  provider_and_id  provider_code  file_type_police  \
count  1.038000e+03     1.038000e+03    1038.000000       1038.000000   
mean   2.019050e+09     3.034275e+10       2.832370          2.845857   
std    2.917011e+04     5.544878e+09       0.554488          0.533666   
min    2.019000e+09     1.201900e+10       1.000000          1.000000   
25%    2.019024e+09     3.201902e+10       3.000000          3.000000   
50%    2.019051e+09     3.201905e+10       3.000000          3.000000   
75%    2.019076e+09     3.201907e+10       3.000000          3.000000   
max    2.019100e+09     3.201910e+10       3.000000          3.000000   

       accident_type  accident_severity  location_accuracy  road_type  \
count    1038.000000        1038.000000        1038.000000     1038.0   
mean        4.936416           2.975915           2.006744        2.0   
std         4.137208           0.159549           0.230200        0.0   
min         1.000000           1.000000           1.000000        2.0   
25%         3.000000           3.000000           2.000000        2.0   
50%         3.000000           3.000000           2.000000        2.0   
75%         7.000000           3.000000           2.000000        2.0   
max        20.000000           3.000000           9.000000        2.0   

        road_shape     day_type  police_unit     one_lane   multi_lane  \
count  1038.000000  1038.000000  1038.000000  1038.000000  1038.000000   
mean      8.958574     3.920039    31.656069     8.494220     0.028902   
std       0.290012     0.456775    18.309148     1.788602     0.321419   
min       4.000000     1.000000    11.000000     0.000000     0.000000   
25%       9.000000     4.000000    12.000000     9.000000     0.000000   
50%       9.000000     4.000000    34.000000     9.000000     0.000000   
75%       9.000000     4.000000    51.000000     9.000000     0.000000   
max      11.000000     4.000000    61.000000     9.000000     5.000000   

       speed_limit  road_intactness   road_width    road_sign   road_light  \
count  1038.000000      1038.000000  1038.000000  1038.000000  1038.000000   
mean      0.093449         0.096339     0.167630     4.839114     8.991329   
std       0.400019         0.405324     0.650519     0.621173     3.002237   
min       0.000000         0.000000     0.000000     1.000000     1.000000   
25%       0.000000         0.000000     0.000000     5.000000     6.000000   
50%       0.000000         0.000000     0.000000     5.000000    11.000000   
75%       0.000000         0.000000     0.000000     5.000000    11.000000   
max       5.000000         4.000000     5.000000     5.000000    11.000000   

       road_control      weather  road_surface  road_object  object_distance  \
count     17.000000  1038.000000   1038.000000  1038.000000      1038.000000   
mean       5.588235     8.426782      8.436416     8.806358         8.934489   
std        1.583462     2.045749      2.003205     0.953134         0.670026   
min        2.000000     1.000000      1.000000     1.000000         1.000000   
25%        5.000000     9.000000      9.000000     9.000000         9.000000   
50%        6.000000     9.000000      9.000000     9.000000         9.000000   
75%        7.000000     9.000000      9.000000     9.000000         9.000000   
max        7.000000     9.000000      9.000000     9.000000         9.000000   

       didnt_cross   cross_mode  cross_location  cross_direction  road1  \
count  1038.000000  1038.000000     1038.000000      1038.000000    0.0   
mean      8.955684     0.129094        0.046243         8.931599    NaN   
std       0.510225     0.903599        0.478849         0.732325    NaN   
min       1.000000     0.000000        0.000000         1.000000    NaN   
25%       9.000000     0.000000        0.000000         9.000000    NaN   
50%       9.000000     0.000000        0.000000         9.000000    NaN   
75%       9.000000     0.000000        0.000000         9.000000    NaN   
max       

In [108]:
m_all[(m_all['street1'].isnull() == False) & (m_all['street1_hebrew'].isnull() == True)]['street1'].value_counts()

0.0       989
9998.0     20
9991.0     16
9992.0      5
202.0       3
139.0       1
5369.0      1
9995.0      1
9994.0      1
107.0       1
Name: street1, dtype: int64

In [29]:
def print_yishuv_street1(street1):
    print(f"street1: {street1}, shape: {m_all[m_all['street1'] == street1].shape}")
    display(m_all[(m_all['street1'] == street1) & (m_all['street1_hebrew'].isnull() == True)]['yishuv_symbol'].value_counts())

In [36]:
print_yishuv_street1(0)

street1: 0, shape: (989, 101)


2730    84
638     63
2720    46
6000    32
7300    31
5000    30
6500    27
2710    20
7400    17
6400    17
4000    16
1139    14
3000    12
637     11
634     11
628     11
8300    10
8800    10
6700     9
654      9
8900     9
7900     9
1061     9
9200     9
2650     8
8000     8
70       8
8700     8
8400     8
7000     7
7100     7
7600     7
7800     7
661      6
1345     6
1290     6
8500     6
7500     6
6900     6
2630     6
1020     6
507      6
531      5
716      5
6100     5
1063     5
2660     5
9100     5
681      5
6600     5
529      4
2400     4
168      4
8600     4
9000     4
2530     4
483      4
494      4
496      4
502      4
481      3
1292     3
473      3
466      3
2640     3
9700     3
2610     3
3567     3
1326     3
504      3
480      3
1304     3
1066     3
2100     3
232      3
720      3
1084     3
1748     3
166      3
705      3
1034     3
6800     3
499      2
498      2
9300     2
7200     2
53       2
6200     2
2043     2
509      2
63       2

In [30]:
print_yishuv_street1(9998)

street1: 9998, shape: (20, 101)


8300    2
126     1
836     1
707     1
8900    1
359     1
7400    1
586     1
4203    1
1264    1
700     1
6800    1
6500    1
365     1
755     1
99      1
345     1
1864    1
2400    1
Name: yishuv_symbol, dtype: int64

In [31]:
print_yishuv_street1(9991)

street1: 9991, shape: (16, 101)


7100    3
3000    2
6400    2
7600    1
2200    1
2650    1
5000    1
7000    1
874     1
166     1
466     1
70      1
Name: yishuv_symbol, dtype: int64

In [32]:
print_yishuv_street1(9992)

street1: 9992, shape: (5, 101)


7100    1
147     1
634     1
3041    1
8500    1
Name: yishuv_symbol, dtype: int64

In [33]:
print_yishuv_street1(202)

street1: 202, shape: (74, 101)


6200    3
Name: yishuv_symbol, dtype: int64

In [35]:
streets = [139, 5369, 9995, 9994, 107]
for street in streets:
    print_yishuv_street1(street)

street1: 139, shape: (75, 101)


577    1
Name: yishuv_symbol, dtype: int64

street1: 5369, shape: (1, 101)


3000    1
Name: yishuv_symbol, dtype: int64

street1: 9995, shape: (1, 101)


1066    1
Name: yishuv_symbol, dtype: int64

street1: 9994, shape: (1, 101)


1748    1
Name: yishuv_symbol, dtype: int64

street1: 107, shape: (102, 101)


807    1
Name: yishuv_symbol, dtype: int64

**Null Conclusion:** Most `street1` == 0, other 49 miscellenious.  
- Some of the miscellenious all the index translation is missing, some only part of the instances translation missing.
- `street1` is not unique in itself, but unique within specific `yishuv_symbol`.  Some `street1` values are missing for specific `yishuv_symbol` and some for many `yishuv_symbol`s.

See details above.


**Specific values mistmatch investigations:**

Didn't do detailed investigation, especially due to all the differences already in the null

## street2 and street2_hebrew

In [109]:
# street2                                      6075 non-null float64
# street2_hebrew                               6072 non-null object

In [110]:
m_all[(m_all['street2'].isnull() == False) & (m_all['street2_hebrew'].isnull() == True)].shape

(3, 101)

In [13]:
m_all[(m_all['street2'].isnull() == False) & (m_all['street2_hebrew'].isnull() == True)]

id  provider_and_id  provider_code  \
201077  2019092408      12019092408              1   
201965  2019099904      12019099904              1   
791490  2019081989      32019081989              3   

                        provider_code_hebrew  file_type_police  accident_type  \
201077  הלשכה המרכזית לסטטיסטיקה - סוג תיק 1               1.0              2   
201965  הלשכה המרכזית לסטטיסטיקה - סוג תיק 1               1.0              5   
791490  הלשכה המרכזית לסטטיסטיקה - סוג תיק 3               3.0             11   

       accident_type_hebrew  accident_severity accident_severity_hebrew  \
201077     התנגשות חזית בצד                  3                      קלה   
201965   התנגשות חזית בחזית                  3                      קלה   
791490                החלקה                  3                      קלה   

         accident_timestamp  location_accuracy location_accuracy_hebrew  \
201077  2019-06-01 11:45:00                  1              עיגון מדויק   
201965  2019-10-10 19:00:00                  1              עיגון מדויק   
791490  2019-06-06 21:30:00                  1              עיגון מדויק   

        road_type road_type_hebrew  road_shape road_shape_hebrew  day_type  \
201077          1    עירונית בצומת           8   כביש ישר / צומת         4   
201965          1    עירונית בצומת           9               אחר         4   
791490          1    עירונית בצומת           9               אחר         4   

       day_type_hebrew  police_unit police_unit_hebrew  one_lane  \
201077         יום אחר           20           מחוז ת"א         2   
201965         יום אחר           20           מחוז ת"א         0   
791490         יום אחר           20           מחוז ת"א         9   

                 one_lane_hebrew  multi_lane            multi_lane_hebrew  \
201077  דו סיטרי + קו הפרדה רצוף           0                          NaN   
201965                       NaN           3  מיפרדה בנויה ללא גדר בטיחות   
791490        לא ידוע מס מסלולים           0                          NaN   

        speed_limit speed_limit_hebrew  road_intactness  \
201077            1         עד 50 קמ"ש                1   
201965            1         עד 50 קמ"ש                1   
791490            0            לא ידוע                0   

       road_intactness_hebrew  road_width road_width_hebrew  road_sign  \
201077            אין ליקויים           4    10.5 עד 14 מטר          3   
201965            אין ליקויים           2       5 עד 7  מטר          3   
791490                לא ידוע           0           לא ידוע          5   

       road_sign_hebrew  road_light road_light_hebrew  road_control  \
201077        אין ליקוי           1      אור יום רגיל           2.0   
201965        אין ליקוי           3   לילה פעלה תאורה           5.0   
791490          לא ידוע           6      לילה לא ידוע           NaN   

       road_control_hebrew  weather weather_hebrew  road_surface  \
201077          רמזור תקין        1           בהיר             1   
201965          תמרור עצור        1           בהיר             1   
791490                 NaN        9        לא ידוע             9   

       road_surface_hebrew  road_object road_object_hebrew  object_distance  \
201077                 יבש            9            לא ידוע                9   
201965                 יבש            9            לא ידוע                9   
791490             לא ידוע            9            לא ידוע                9   

       object_distance_hebrew  didnt_cross didnt_cross_hebrew  cross_mode  \
201077                לא ידוע            9            לא ידוע           0   
201965                לא ידוע            9            לא ידוע           0   
791490                לא ידוע            9            לא ידוע           0   

       cross_mode_hebrew  cross_location cross_location_hebrew  ...  \
201077               NaN               0                   NaN  ...   
201965               NaN               0                   NaN  ...   
791490               NaN               0      

In [111]:
m_all[(m_all['street2'].isnull() == False) & (m_all['street2_hebrew'].isnull() == True)].describe()

id  provider_and_id  provider_code  file_type_police  \
count  3.000000e+00     3.000000e+00       3.000000          3.000000   
mean   2.019091e+09     1.868576e+10       1.666667          1.666667   
std    8.997155e+03     1.154700e+10       1.154701          1.154701   
min    2.019082e+09     1.201909e+10       1.000000          1.000000   
25%    2.019087e+09     1.201910e+10       1.000000          1.000000   
50%    2.019092e+09     1.201910e+10       1.000000          1.000000   
75%    2.019096e+09     2.201909e+10       2.000000          2.000000   
max    2.019100e+09     3.201908e+10       3.000000          3.000000   

       accident_type  accident_severity  location_accuracy  road_type  \
count       3.000000                3.0                3.0        3.0   
mean        6.000000                3.0                1.0        1.0   
std         4.582576                0.0                0.0        0.0   
min         2.000000                3.0                1.0        1.0   
25%         3.500000                3.0                1.0        1.0   
50%         5.000000                3.0                1.0        1.0   
75%         8.000000                3.0                1.0        1.0   
max        11.000000                3.0                1.0        1.0   

       road_shape  day_type  police_unit  one_lane  multi_lane  speed_limit  \
count    3.000000       3.0          3.0  3.000000    3.000000     3.000000   
mean     8.666667       4.0         20.0  3.666667    1.000000     0.666667   
std      0.577350       0.0          0.0  4.725816    1.732051     0.577350   
min      8.000000       4.0         20.0  0.000000    0.000000     0.000000   
25%      8.500000       4.0         20.0  1.000000    0.000000     0.500000   
50%      9.000000       4.0         20.0  2.000000    0.000000     1.000000   
75%      9.000000       4.0         20.0  5.500000    1.500000     1.000000   
max      9.000000       4.0         20.0  9.000000    3.000000     1.000000   

       road_intactness  road_width  road_sign  road_light  road_control  \
count         3.000000         3.0   3.000000    3.000000       2.00000   
mean          0.666667         2.0   3.666667    3.333333       3.50000   
std           0.577350         2.0   1.154701    2.516611       2.12132   
min           0.000000         0.0   3.000000    1.000000       2.00000   
25%           0.500000         1.0   3.000000    2.000000       2.75000   
50%           1.000000         2.0   3.000000    3.000000       3.50000   
75%           1.000000         3.0   4.000000    4.500000       4.25000   
max           1.000000         4.0   5.000000    6.000000       5.00000   

        weather  road_surface  road_object  object_distance  didnt_cross  \
count  3.000000      3.000000          3.0              3.0          3.0   
mean   3.666667      3.666667          9.0              9.0          9.0   
std    4.618802      4.618802          0.0              0.0          0.0   
min    1.000000      1.000000          9.0              9.0          9.0   
25%    1.000000      1.000000          9.0              9.0          9.0   
50%    1.000000      1.000000          9.0              9.0          9.0   
75%    5.000000      5.000000          9.0              9.0          9.0   
max    9.000000      9.000000          9.0              9.0          9.0   

       cross_mode  cross_location  cross_direction  road1  road2   km  km_raw  \
count         3.0             3.0              3.0    0.0    0.0  0.0     0.0   
mean          0.0             0.0              9.0    NaN    NaN  NaN     NaN   
std           0.0             0.0              0.0    NaN    NaN  NaN     NaN   
min           0.0             0.0              9.0    NaN    NaN  NaN     NaN   
25%           0.0             0.0              9.0    NaN    NaN  NaN     NaN   
50%           0.0             0.0              9.0    NaN    NaN  NaN     NaN   
75%           0.0             0.0              9.0    NaN    

In [115]:
m_all[m_all['street2'] == 202].shape

(14, 101)

In [12]:
m_all[m_all['street2'] == 202][['yishuv_symbol','street2','street2_hebrew','accident_year']].sort_values(by='yishuv_symbol')

yishuv_symbol  street2 street2_hebrew  accident_year
195679             70    202.0    שפירא אברהם           2019
196390           2400    202.0     יהדות קנדה           2019
200036           2400    202.0     יהדות קנדה           2019
763236           2400    202.0     יהדות קנדה           2019
788523           2400    202.0     יהדות קנדה           2019
6473             3570    202.0   דרך הנחשונים           2019
201077           6200    202.0            NaN           2019
201965           6200    202.0            NaN           2019
791490           6200    202.0            NaN           2019
203750           6700    202.0         אלחדיף           2019
202263           8500    202.0          תל חי           2019
197027           9000    202.0    אברהם אבינו           2019
196292           9600    202.0          הגליל           2019
199682           9600    202.0          הגליל           2019

**Conclusion:** `street2` is not unique on it's own, but only with combination with `yishuv_symbol`.  Seems that there is an issue with `street2` == 202 and `yishuv_symbol` == 6200

**Specific values mistmatch investigations:**

Didn't do detailed investigation, especially due to all the differences already in the null

### non_urban_intersection and non_urban_intersection_hebrew

In [120]:
# non_urban_intersection                       4309 non-null float64
# non_urban_intersection_hebrew                4302 non-null object

In [121]:
m_all[(m_all['non_urban_intersection'].isnull() == False) & (m_all['non_urban_intersection_hebrew'].isnull() == True)].shape

(7, 101)

In [122]:
m_all[(m_all['non_urban_intersection'].isnull() == False) & (m_all['non_urban_intersection_hebrew'].isnull() == True)]

id  provider_and_id  provider_code  \
5924    2019025840      12019025840              1   
202628  2019004747      12019004747              1   
782328  2019056033      32019056033              3   
782329  2019056033      32019056033              3   
787592  2019071126      32019071126              3   
796815  2019095698      32019095698              3   
796816  2019095698      32019095698              3   

                        provider_code_hebrew  file_type_police  accident_type  \
5924    הלשכה המרכזית לסטטיסטיקה - סוג תיק 1               1.0              2   
202628  הלשכה המרכזית לסטטיסטיקה - סוג תיק 1               1.0              2   
782328  הלשכה המרכזית לסטטיסטיקה - סוג תיק 3               3.0              4   
782329  הלשכה המרכזית לסטטיסטיקה - סוג תיק 3               3.0              4   
787592  הלשכה המרכזית לסטטיסטיקה - סוג תיק 3               3.0              3   
796815  הלשכה המרכזית לסטטיסטיקה - סוג תיק 3               3.0              3   
796816  הלשכה המרכזית לסטטיסטיקה - סוג תיק 3               3.0              3   

       accident_type_hebrew  accident_severity accident_severity_hebrew  \
5924       התנגשות חזית בצד                  3                      קלה   
202628     התנגשות חזית בצד                  3                      קלה   
782328       התנגשות צד בצד                  3                      קלה   
782329       התנגשות צד בצד                  3                      קלה   
787592   התנגשות חזית באחור                  3                      קלה   
796815   התנגשות חזית באחור                  3                      קלה   
796816   התנגשות חזית באחור                  3                      קלה   

         accident_timestamp  location_accuracy location_accuracy_hebrew  \
5924    2019-02-27 12:15:00                  1              עיגון מדויק   
202628  2019-09-10 09:30:00                  1              עיגון מדויק   
782328  2019-05-16 16:00:00                  1              עיגון מדויק   
782329  2019-05-16 16:00:00                  1              עיגון מדויק   
787592  2019-02-12 18:15:00                  1              עיגון מדויק   
796815  2019-05-03 16:00:00                  1              עיגון מדויק   
796816  2019-05-03 16:00:00                  1              עיגון מדויק   

        road_type  road_type_hebrew  road_shape road_shape_hebrew  day_type  \
5924            3  לא-עירונית בצומת           8   כביש ישר / צומת         4   
202628          3  לא-עירונית בצומת           9               אחר         4   
782328          3  לא-עירונית בצומת           9               אחר         4   
782329          3  לא-עירונית בצומת           9               אחר         4   
787592          3  לא-עירונית בצומת           9               אחר         4   
796815          3  לא-עירונית בצומת           9               אחר         4   
796816          3  לא-עירונית בצומת           9               אחר         4   

       day_type_hebrew  police_unit police_unit_hebrew  one_lane  \
5924           יום אחר           34           מרחב נגב         0   
202628         יום אחר           34           מרחב נגב         9   
782328         יום אחר           14         מרחב עמקים         9   
782329         יום אחר           14         מרחב עמקים         9   
787592         יום אחר           52          מרחב שפלה         9   
796815         יום אחר           14         מרחב עמקים         9   
796816         יום אחר           14         מרחב עמקים         9   

           one_lane_hebrew  multi_lane            multi_lane_hebrew  \
5924                   NaN           3  מיפרדה בנויה ללא גדר בטיחות   
202628  לא ידוע מס מסלולים           0                          NaN   
782328  לא ידוע מס מסלולים           0                          NaN   
782329  לא ידוע מס מסלולים           0                          NaN   
787592  לא ידוע מס מסלולים           0                          NaN   
796815  לא ידוע מס מסלולים           0                          NaN   
796816  לא ידוע מס מסלולים           0                   

In [123]:
m_all[(m_all['non_urban_intersection'].isnull() == False) & (m_all['non_urban_intersection_hebrew'].isnull() == True)]['non_urban_intersection'].value_counts()

1077.0    4
8015.0    2
4186.0    1
Name: non_urban_intersection, dtype: int64

In [125]:
m_all[m_all['non_urban_intersection'] == 1077].shape

(4, 101)

In [16]:
m_all[m_all['non_urban_intersection'] == 1077][['non_urban_intersection','non_urban_intersection_hebrew','non_urban_intersection_by_junction_number','km','road1','road2','provider_and_id']]

non_urban_intersection non_urban_intersection_hebrew  \
782328                  1077.0                           NaN   
782329                  1077.0                           NaN   
796815                  1077.0                           NaN   
796816                  1077.0                           NaN   

       non_urban_intersection_by_junction_number     km  road1  road2  \
782328                          צומת נצרת (דרום)  171.0   75.0    NaN   
782329                          צומת נצרת (דרום)  171.0   75.0    NaN   
796815                          צומת נצרת (דרום)  171.0   75.0    NaN   
796816                          צומת נצרת (דרום)  171.0   75.0    NaN   

        provider_and_id  
782328      32019056033  
782329      32019056033  
796815      32019095698  
796816      32019095698

In [126]:
m_all[m_all['non_urban_intersection'] == 8015].shape

(2, 101)

In [17]:
m_all[m_all['non_urban_intersection'] == 8015][['non_urban_intersection','non_urban_intersection_hebrew','non_urban_intersection_by_junction_number','km','road1','road2','provider_and_id']]

non_urban_intersection non_urban_intersection_hebrew  \
5924                    8015.0                           NaN   
202628                  8015.0                           NaN   

       non_urban_intersection_by_junction_number   km  road1  road2  \
5924                        יישוב באר שבע (לערד)  0.0   60.0    NaN   
202628                      יישוב באר שבע (לערד)  0.0   60.0    NaN   

        provider_and_id  
5924        12019025840  
202628      12019004747

In [127]:
m_all[m_all['non_urban_intersection'] == 4186].shape

(3, 101)

In [15]:
m_all[m_all['non_urban_intersection'] == 4186][['non_urban_intersection','non_urban_intersection_hebrew','non_urban_intersection_by_junction_number','km','road1','road2','provider_and_id']]

non_urban_intersection non_urban_intersection_hebrew  \
767096                  4186.0                 מחלף נס ציונה   
787592                  4186.0                           NaN   
791105                  4186.0                 מחלף נס ציונה   

       non_urban_intersection_by_junction_number     km  road1   road2  \
767096                             מחלף נס ציונה   65.0  423.0   431.0   
787592                             מחלף נס ציונה  160.0  431.0  4313.0   
791105                             מחלף נס ציונה  160.0  431.0   423.0   

        provider_and_id  
767096      32019012216  
787592      32019071126  
791105      32019080980

**Null conclusions** of `non_urban_intersection`:  It's mostly our internal issue since in all cases `non_urban_intersection_by_junction_number` is not empty.
 Issues: All values of 1077, 8015, some values of 4186

In [129]:
m_all['non_urban_intersection'].nunique()

797

**Specific values conclusion:** because of issues above, and so many unique values, not investigated

## accident_hour_raw and accident_hour_raw_hebrew

In [131]:
# accident_hour_raw                            53030 non-null int64
# accident_hour_raw_hebrew                     53030 non-null object

In [132]:
calc_diff_counts_hebrew(m_all, 'accident_hour_raw')

Shape of data: (53030, 101)


0     0
1     0
2     0
3     0
4     0
5     0
6     0
7     0
8     0
9     0
10    0
11    0
12    0
13    0
14    0
15    0
16    0
17    0
18    0
19    0
20    0
21    0
22    0
23    0
24    0
25    0
26    0
27    0
28    0
29    0
30    0
31    0
32    0
33    0
34    0
35    0
36    0
37    0
38    0
39    0
40    0
41    0
42    0
43    0
44    0
45    0
46    0
47    0
48    0
49    0
50    0
51    0
52    0
53    0
54    0
55    0
56    0
57    0
58    0
59    0
60    0
61    0
62    0
63    0
64    0
65    0
66    0
67    0
68    0
69    0
70    0
71    0
72    0
73    0
74    0
75    0
76    0
77    0
78    0
79    0
80    0
81    0
82    0
83    0
84    0
85    0
86    0
87    0
88    0
89    0
90    0
91    0
92    0
93    0
94    0
95    0
dtype: int64